hoot-core/src/main/cpp/hoot/core/conflate/DiffConflator.cpp

```diff
@@ -355,80 +452,144 @@ long DiffConflator::_snapSecondaryRoadsBackToRef()
-            // poi/poly is the only conflation type that allows intra-dataset matches. We don't want
-            // these to be removed from the diff output.
-            !(match->getName() == PoiPolygonMatch::MATCH_NAME &&
-              _intraDatasetMatchOnlyElementIds.contains(pit->first)))
-        {
-          LOG_TRACE("Removing element involved in match: " << pit->first << "...");
-          RecursiveElementRemover(pit->first).apply(_map);
-        }
-        if (e2 &&
-            e2->getStatus() == status &&
-            // see related comment above
-            (status != Status::Unknown1 || notSnappedCrit->isSatisfied(e2)) &&
-            // see related comment above
-            !(match->getName() == PoiPolygonMatch::MATCH_NAME &&
-             _intraDatasetMatchOnlyElementIds.contains(pit->second)))
-        {
-          LOG_TRACE("Removing element involved in match: " << pit->second << "...");
-          RecursiveElementRemover(pit->second).apply(_map);
-        }
+void DiffConflator::_removeMatchElementPairCompletely(
+  const ConstMatchPtr& match, const std::pair<ElementId, ElementId>& elementPair,
+  const Status& status)
+{
+  LOG_TRACE(
+    "Removing match element pair completely: " << elementPair << " for elements with status: " <<
+    status << "...");
+  LOG_VART(match->getName());
+  const MatchType matchType = match->getType();
+  LOG_VART(matchType);
+  LOG_VART(match->getClassification());
+  LOG_VART(match->getMatchPairs().size());
+
+  if (!elementPair.first.isNull())
+  {
+    ElementPtr e1 = _map->getElement(elementPair.first);
+    if (e1)
+    {
+      QString msg =
+        "Removed entire element: " + e1->getElementId().toString() + " with status: " +
+        e1->getStatus().toString() + ", involved in match of type: " + matchType.toString() + "...";
+      if (_satisfiesCompleteElementRemovalCondition(e1, status, match))
+      {
+        RecursiveElementRemover(e1->getElementId()).apply(_map);
+      }
+      else
+      {
+        msg = msg.replace("Removed", "Did not remove");
+      }
+      LOG_TRACE(msg);
+    }
+  }
+  if (!elementPair.second.isNull())
+  {
+    ElementPtr e2 = _map->getElement(elementPair.second);
+    if (e2)
+    {
+      QString msg =
+        "Removed entire element: " + e2->getElementId().toString() + " with status: " +
+        e2->getStatus().toString() + ", involved in match of type: " + matchType.toString() + "...";
+      if (_satisfiesCompleteElementRemovalCondition(e2, status, match))
+      {
+        RecursiveElementRemover(e2->getElementId()).apply(_map);
+      }
+      else
+      {
+        msg = msg.replace("Removed", "Did not remove");
       }
+      LOG_TRACE(msg);
     }
   }
+}
 
-  OsmMapWriterFactory::writeDebugMap(_map, "after-removing-" + status.toString() + "-matches");
+void DiffConflator::_removePartialSecondaryMatchElements()
+{
+  std::vector<MergerPtr> relationMergers;
+  _createMergers(relationMergers);
+  _mergeFeatures(relationMergers);
 }
```

Comment:
The partial feature removal ended up being best accomplished using the same merging framework that Unifying uses.